### PR TITLE
Renderer class

### PR DESCRIPTION
**Pull request by memory-hunter**
Created separate Renderer class and added multi-mesh rendering as a demo to see that it works

### CODE REVIEW

File: src/main/java/core/Engine.java [hunk 1]
```diff
@@ -3,14 +3,15 @@
 import io.Input;
 import utils.TimeUtils;
 
+import java.util.ArrayList;
+
 public class Engine{
 
     public Window window;
-    public Entity entity;
+    public ArrayList<Entity> entities = new ArrayList<Entity>();
```
Comment (DerWaschbar) [inline src/main/java/core/Engine.java:11]:
```suggestion
    public List<Entity> entities = new ArrayList<Entity>();
```
Any specific reason to use `ArrayList`?
Use abstract interfaces if concrete implementation is not required.

File: src/main/java/core/Engine.java [hunk 2]
```diff
@@ -23,7 +24,10 @@ public void init(){
         TimeUtils.init(60);
         Input.init();
         window.init();
-        entity.init();
+        for (Entity entity : entities) {
+            entity.init();
+        }
```
Comment (DerWaschbar) [inline src/main/java/core/Engine.java:29]: Try using streams: `forEach`.

File: src/main/java/core/Engine.java [hunk 3]
```diff
@@ -53,18 +57,26 @@ private boolean update(){
         TimeUtils.updateFps();
         TimeUtils.updateCycle();
         window.update();
-        entity.update();
+        for (Entity entity : entities) {
+            entity.update();
+        }
```
Comment (DerWaschbar) [inline src/main/java/core/Engine.java:62]: Try using streams: `forEach`.

File: src/main/java/core/Engine.java [hunk 4]
```diff
@@ -53,18 +57,26 @@ private boolean update(){
...
         return true;
     }
 
     private void render() {
         window.clear();
-        entity.render();
+        for (Entity entity : entities) {
+            entity.render();
+        }
```
Comment (DerWaschbar) [inline src/main/java/core/Engine.java:70]: Try using streams: `forEach`.

File: src/main/java/core/Engine.java [hunk 5]
```diff
@@ -53,18 +57,26 @@ private boolean update(){
...
+        Renderer.renderAll();
         window.render();
     }
 
     public void stop(){
-        entity.destroy();
+        for (Entity entity : entities) {
+            entity.destroy();
+        }
```
Comment (DerWaschbar) [inline src/main/java/core/Engine.java:78]: Try using streams: `forEach`.

File: src/main/java/core/Renderer.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
+package core;
+
+import graphics.ShaderProgram;
+
+import java.util.ArrayList;
+
+public class Renderer {
+    private static final ArrayList<Item> items = new ArrayList<Item>();
```
Comment (DerWaschbar) [inline src/main/java/core/Renderer.java:8]:
```suggestion
    private static final List<Item> items = new ArrayList<Item>();
```
Same as before.

File: src/main/java/core/Renderer.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    private static ShaderProgram shaderProgram;
+
+    private Renderer() {
+    }
+
+    public static void initAll() {
+        for (Item item : items) {
+            item.init();
+        }
```
Comment (DerWaschbar) [inline src/main/java/core/Renderer.java:17]: Streams?

File: src/main/java/core/Renderer.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public static void renderAll() {
+        if(items.size() < 1) {
+            System.out.println("No items to render");
+        }
+        shaderProgram.bind();
+        for (Item item : items) {
+            item.render();
+        }
```
Comment (DerWaschbar) [inline src/main/java/core/Renderer.java:39]: Streams?

**Review by DerWaschbar — DISMISSED** [review]

File: src/main/java/core/Renderer.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class Renderer {
+    private static final List<Item> items = new ArrayList<>();
+
+    private Renderer() {
+    }
+
+    public static void initAll() {
```
Comment (giokepa) [inline src/main/java/core/Renderer.java:14]: Does not need to name initAll you could just

```
public static void init() {}

```

File: src/main/java/core/Renderer.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        items.forEach(Item::init);
+    }
+
+    public static void addItem(Item item) {;
+        if(!items.contains(item)) {
+            items.add(item);
+        }
+    }
+
+    public static void removeItem(Item item) {
```
Comment (giokepa) [inline src/main/java/core/Renderer.java:24]: what if item does not exist?

File: src/main/java/core/Renderer.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        items.remove(item);
+    }
+
+    public static void renderAll() {
```
Comment (giokepa) [inline src/main/java/core/Renderer.java:28]: name it
``` public static void render() {} ```

**Comment by memory-hunter** [comment]
Made more changes

**Review by giokepa — APPROVED** [review]
Every issue was corrected

**Pull request merged**